server/lyft/checks/github_client.go

```diff
@@ -3,21 +3,163 @@ package checks
 import (
 	"context"
 	"fmt"
+	"strconv"
+	"strings"
 
+	"github.com/google/go-github/v31/github"
+	"github.com/pkg/errors"
+	"github.com/runatlantis/atlantis/server/core/db"
+	"github.com/runatlantis/atlantis/server/events/models"
 	"github.com/runatlantis/atlantis/server/events/vcs"
 	"github.com/runatlantis/atlantis/server/events/vcs/types"
 	"github.com/runatlantis/atlantis/server/logging"
 	"github.com/runatlantis/atlantis/server/lyft/feature"
 )
 
+// Reference: https://github.com/github/docs/issues/3765
+const maxChecksOutputLength = 65535
+
+// github checks status
+type CheckStatus int
+
+const (
+	Queued CheckStatus = iota
+	InProgress
+	Completed
+)
+
+func (e CheckStatus) String() string {
+	switch e {
+	case Queued:
+		return "queued"
+	case InProgress:
+		return "in_progress"
+	case Completed:
+		return "completed"
+	}
+	return ""
+}
+
+// github checks conclusion
+type ChecksConclusion int
+
+const (
+	Neutral ChecksConclusion = iota
+	TimedOut
+	ActionRequired
+	Cancelled
+	Failure
+	Success
+)
+
+func (e ChecksConclusion) String() string {
+	switch e {
+	case Neutral:
+		return "neutral"
+	case TimedOut:
+		return "timed_out"
+	case ActionRequired:
+		return "action_required"
+	case Cancelled:
+		return "cancelled"
+	case Failure:
+		return "failure"
+	case Success:
+		return "success"
+	}
+	return ""
+}
+
 // [WENGINES-4643] TODO: Remove this wrapper and add checks implementation to UpdateStatus() directly after github checks is stable
 type ChecksClientWrapper struct {
 	*vcs.GithubClient
 	FeatureAllocator feature.Allocator
 	Logger           logging.Logger
+	Db               *db.BoltDB
 }
 
 func (c *ChecksClientWrapper) UpdateStatus(ctx context.Context, request types.UpdateStatusRequest) error {
+
+	if !c.isChecksEnabled(ctx, request) {
+		return c.GithubClient.UpdateStatus(ctx, request)
+	}
+
+	// Project level policy check command status is set to either success or failure directly
```

Comment:
? can't we just create a pending run so we don't have to worry about this extra logic here?

Reply:
Hmm yeah I think that would make it simpler